server.coffee

```diff
@@ -12,6 +12,7 @@ configWrapper = require './lib/configWrapper'
 load = require './lib/load'
 
 MODULES = config.modules
+ALLOW_ORIGIN_HEADERS = config.httpHeaders
```

Comment:
You are setting `ALLOW_ORIGIN_HEADERS` to the list of all headers here, which doesn't make sense.  You are then later changing it to be a completely different thing (the value of just the `allowOrigin` header).